PGO profiling should take Speedometer3 into account.

dewei-zhu opened the pull request:
#### 3e68b1d06e4d8f320e7964ba7b316473ea704b80
<pre>
PGO profiling should take Speedometer3 into account.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261689">https://bugs.webkit.org/show_bug.cgi?id=261689</a>
rdar://111352676

Reviewed by Yusuke Suzuki and Wenson Hsieh.

Inlcude Speedometer3 into PGO profile collection process.
Assigning 20% weight to Speedometer3 and reduce Speedometer2 weight from
60% to 40% so that overall Speeodmeter famility weight stays unchanged.
New weight is derived from experiment which tried couple different formulas.

* Tools/Scripts/build-and-collect-pgo-profiles:
* Tools/Scripts/pgo-profile:

Canonical link: <a href="https://commits.webkit.org/268104@main">https://commits.webkit.org/268104@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1b866b3d54f0e09fce1539f78843338f8f6f197b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18700 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19041 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19644 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20564 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17506 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18897 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22350 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19182 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18925 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19074 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16282 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21442 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16297 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17042 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23493 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17322 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17214 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21387 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17815 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16874 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4439 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21241 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17654 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/Scripts/build-and-collect-pgo-profiles:
```diff
@@ -124,15 +134,18 @@ SPTH='OpenSource/Tools/Scripts'
 rm *.result
 
-$SPTH/pgo-profile combine --jetstream "$BASE/jetstream" --speedometer "$BASE/speedometer" --motionmark "$BASE/motionmark" --output "$BASE/output"
+$SPTH/pgo-profile combine --jetstream "$BASE/jetstream" --speedometer2 "$BASE/speedometer2" ---speedometer3 "$BASE/speedometer3" --motionmark "$BASE/motionmark" --output "$BASE/output"
```
Is `---speedometer3` (3 dashes) intentional?

dewei-zhu replied:
Nice catch. I've updated the patch accordingly.

Constellation reviewed (approved):
r=me

dewei-zhu committed:
PGO profiling should take Speedometer3 into account.
https://bugs.webkit.org/show_bug.cgi?id=261689
rdar://111352676

Reviewed by Yusuke Suzuki and Wenson Hsieh.

Inlcude Speedometer3 into PGO profile collection process.
Assigning 20% weight to Speedometer3 and reduce Speedometer2 weight from
60% to 40% so that overall Speeodmeter famility weight stays unchanged.
New weight is derived from experiment which tried couple different formulas.

* Tools/Scripts/build-and-collect-pgo-profiles:
* Tools/Scripts/pgo-profile:

Canonical link: https://commits.webkit.org/268104@main
